x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MlMetadata.java

```diff
@@ -64,7 +64,7 @@ public boolean isResetMode() {
 
     @Override
     public Version getMinimalSupportedVersion() {
-        return Version.CURRENT.minimumIndexCompatibilityVersion();
+        return Version.CURRENT.minimumCompatibilityVersion();
```

Comment:
@droberts195 I've made some changes for ML too. The `minimalSupportedVersion` refers to wire-compatibility, so it should never be `minimumIndexCompatibilityVersion` (as the index BWC guarantees are wider than the wire ones).
Let me know if this doesn't make sense to change as part of this effort?